test/unit/buffer_spec.lua

```diff
@@ -211,93 +210,244 @@ describe('buffer functions', function()
-
-      eq(14, width)
-      eq("this is a test", helpers.ffi.string(output_buffer, width))
-
-    end)
-
-    it('should print no file name', function()
-      local width = build_stl_str_hl("%f")
-
-      eq(9, width)
-      eq("[No Name]", helpers.ffi.string(output_buffer, width))
-
-    end)
-
-    it('should print the relative file name', function()
-      buffer.setfname(globals.curbuf, to_cstr("Makefile"), NULL, 1)
-      local width = build_stl_str_hl("%f")
-
-      eq(8, width)
-      eq("Makefile", helpers.ffi.string(output_buffer, width))
-
-    end)
-
-    it('should print the full file name', function()
-      buffer.setfname(globals.curbuf, to_cstr("Makefile"), NULL, 1)
-
-      local width = build_stl_str_hl("%F")
-
-      assert.is_true(8 < width)
-      neq(NULL, string.find(helpers.ffi.string(output_buffer, width), "Makefile"))
-
-    end)
-
-    it('should print the tail file name', function()
-      buffer.setfname(globals.curbuf, to_cstr("src/nvim/buffer.c"), NULL, 1)
-
-      local width = build_stl_str_hl("%t")
-
-      eq(8, width)
-      eq("buffer.c", helpers.ffi.string(output_buffer, width))
-
-    end)
-
-    it('should print the buffer number', function()
-      buffer.setfname(globals.curbuf, to_cstr("src/nvim/buffer.c"), NULL, 1)
-
-      local width = build_stl_str_hl("%n")
-
-      eq(1, width)
-      eq("1", helpers.ffi.string(output_buffer, width))
-    end)
-
-    it('should print the current line number in the buffer', function()
-      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
-
-      local width = build_stl_str_hl("%l")
-
-      eq(1, width)
-      eq("0", helpers.ffi.string(output_buffer, width))
-
-    end)
-
-    it('should print the number of lines in the buffer', function()
-      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
-
-      local width = build_stl_str_hl("%L")
+    -- Use this function to simplify testing the comparison between
+    --  the format string and the resulting statusline.
+    --
+    -- @param description The description of what the test should be doing
+    -- @param statusline_cell_count The number of cells available in the statusline
+    -- @param input_stl The format string for the statusline
+    -- @param expected_stl The expected result string for the statusline
+    --
+    -- Options can be placed in an optional dictionary as the last parameter
+    -- @option
+    -- @option expected_cell_count The expected number of cells build_stl_str_hl will return
+    -- @option  expected_byte_length The expected byte length of the string
+    local function statusline_test (description,
+                                    statusline_cell_count,
+                                    input_stl,
+                                    expected_stl,
+                                    arg)
+
+      -- arg is our optional parameter
+      arg = arg or {}
+
+      local fillchar = arg.fillchar or (' '):byte()
```

Comment:
It should be mentioned in the function comment that arg can have a key `fillchar`.